documentation/design-docs/diagnostics-client-library.md

```diff
@@ -240,6 +240,110 @@ public static int AttachProfiler(int processId, Guid profilerGuid, string profil
 }
 ```
 
+#### 8. Write a simple monitor agent
+
+This sample shows how to use the `DiagnosticsAgent` API to create a simple Agent application.
+```cs
+namespace Microsoft.Diagnostics.Tools.Reverse
+{
+    class Program
+    {
+        static ConcurrentDictionary<Guid, (Task, EventPipeSession)> sessionDict = new ConcurrentDictionary<int, (Task, EventPipeSession)>();
+        static ConcurrentDictionary<Guid, (int, DiagnosticsClient)> clientDict = new ConcurrentDictionary<int, (int, DiagnosticsClient)>();
+
+        static void Main(string[] args)
+        {
+            // get the address for the server (this sample assumes Linux)
+            string address = (args.Length >= 1) ? args[0] : "~/myawesome.sock";
+            try
+            {
+                using (DiagnosticsAgent agent = new DiagnosticsAgent(address))
+                {
+                    agent.OnDiagnosticsConnection += (sender, eventArgs) =>
```

Comment:
Can we make this more of a listener/socket pull style API? 

```C#
var agent = new DiagnosticsAgent(address);
while (true)
{ 
   var connection = await agent.AcceptAsync();
   _ = ProcessConnectionAsync(connection);
}
```

Reply:
Would `DiagnosticsAgent.AcceptAsync()` return a `connection` _every time_ a runtime instance connects to the agent or only the first time?  The current model raises the `OnDiagnosticsConnection` event on only the first connection and users are expected to reuse the `DiagnosticsClient` (which abstracts subsequent reconnects) they get just like they use the traditional `DiagnosticsClient` today.  if we change it to be "every time an instance connects" we may want to modify the `DiagnosticsClient` API for dispatching commands, or not use it entirely.

Reply:
I'd suggest it is _first time_ only. I am hoping to preserve the model that you get a DiagnosticClient and then you send an arbitrary number of commands without making the user aware of the underlying transport stream management.